R/include-gld-purchases.R

```diff
@@ -0,0 +1,39 @@
+#' Include only those who have a purchase of a glucose lowering drug (GLD).
+#'
+#' See [algorithm] for the logic used to filter these patients.
+#'
+#' @return The same type as the input data, default as a [tibble::tibble()].
+#' @keywords internal
+#'
+#' @examples
+#' \dontrun{
+#' register_data$lmdb |> include_gld_purchases()
+#' }
+include_gld_purchases <- function(data) {
+  verify_required_variables(data, "lmdb")
+  criteria <- get_algorithm_logic("gld") |>
+    # To convert the string into an R expression.
+    rlang::parse_expr()
+  data |>
+    column_names_to_lower() |>
+    # Use !! to inject the expression into filter.
+    dplyr::filter(!!criteria) |>
+    # Keep only the columns we need.
```

Comment:
As mentioned in the comment on the test below: the function only needs either of "name" or "vnr" to work, so we should identify which one is present in the input and use that one (and a default if both are present, probably "name"). The exclude_wld_purchases() function that does the actual censoring/exclusion of weightloss drugs based on "name" or "vnr" also needs to assert which one is passed to it and handle it accordingly.

Reply:
Wait, could you explain, why does it need one or the other? Can we decide which we want and require that? Does LMDB not always have these variables?

Reply:
Yes, it needs one or the other, and I think both are equally accessible in LMDB, and it only depends on if the user has requested the variables. E.g. In my PhD, I had "name", but not "vnr", but on DARTER, we have "vnr", but not "name".
If we had to choose, "name" is much easier to maintain than "vnr" (vnr numbers get added over the years when packagings change, but brand names stay the same).

Reply:
Let's just require they have `name` then? If they don't have it, to request it, what do you think?

Reply:
name and vnr no longer needed, problem solved.